share/transform/chain/links.py

```diff
@@ -484,6 +489,9 @@ def __init__(self, parser):
         super().__init__()
 
     def execute(self, obj):
+        # callable will return True for classes as well as functions
+        if isinstance(self._parser, types.FunctionType):
+            self._parser = self._parser(obj)
```

Comment:
After team discussion, this seemed to cause some issues with the agent_parser. Will send back to additional development until updated.